opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -194,24 +209,41 @@ def merge(self, other: "Resource") -> "Resource":
         """
         merged_attributes = self.attributes
         merged_attributes.update(other.attributes)
-        return Resource(merged_attributes)
+
+        if self.schema_url == "":
+            schema_url = other.schema_url
+        elif other.schema_url == "":
+            schema_url = self.schema_url
+        elif self.schema_url == other.schema_url:
+            schema_url = other.schema_url
+        else:
+            schema_url = "merge-conflict"
+            logger.error(
+                "Failed to merge resources: The Schema URL of the old and updating resources are not empty and are different"
+            )
+
+        return Resource(merged_attributes, schema_url)
 
     def __eq__(self, other: object) -> bool:
         if not isinstance(other, Resource):
             return False
-        return self._attributes == other._attributes
+        return (
+            self._attributes == other._attributes
+            and self._schema_url == other._schema_url
+        )
 
     def __hash__(self):
-        return hash(dumps(self._attributes, sort_keys=True))
+        return hash(dumps(self._attributes, sort_keys=True) + self._schema_url)
```

Comment:
```suggestion
        return hash(dumps(self._attributes, sort_keys=True) + "|" + self._schema_url)
```

When combining multiple values to create a hash/unique identifier I always include a delimiter to ensure that `"a"+"b"` can be distinguished from `"ab"+""` and `""+"ab"`.

Reply:
That's a great idea!

Reply:
Done. I used a [slightly different approach](32ce84880e40d7f33b6736ba54affbf5a14085b6) at first, but this is simpler.

Reply:
The delimeter implementation actually doesn't pass the benchmark test. It doesn't seem like it'd have that much of a performance penalty to concatenate an extra string, but maybe the extra memory copy just isn't worth it. I went back to my implementation for now, and hopefully that passes.